pkg/tnf/operator/starter.go

```diff
@@ -56,24 +58,43 @@ func HandleDualReplicaClusters(
 	runTnfResourceController(ctx, controllerContext, kubeClient, dynamicClient, operatorClient, kubeInformersForNamespaces)
 
 	// we need node names for assigning auth jobs to specific nodes
+	var once sync.Once
 	klog.Infof("watching for nodes...")
 	_, err := controlPlaneNodeInformer.AddEventHandler(cache.ResourceEventHandlerFuncs{
 		AddFunc: func(obj interface{}) {
-			node, ok := obj.(*corev1.Node)
+			_, ok := obj.(*corev1.Node)
 			if !ok {
 				klog.Warningf("failed to convert node to Node %+v", obj)
 			}
-			runTnfAuthJobController(ctx, node.GetName(), controllerContext, operatorClient, kubeClient, kubeInformersForNamespaces)
-			runTnfAfterSetupJobController(ctx, node.GetName(), controllerContext, operatorClient, kubeClient, kubeInformersForNamespaces)
+			// ensure we have both control plane nodes before creating jobs
+			nodeList, err := kubeClient.CoreV1().Nodes().List(ctx, metav1.ListOptions{
+				LabelSelector: "node-role.kubernetes.io/master",
+			})
+			if err != nil {
+				klog.Errorf("failed to list nodes while waiting to create TNF jobs: %v", err)
+				return
+			}
+			if len(nodeList.Items) != 2 {
+				klog.Info("not starting TNF jobs yet, waiting for 2 control plane nodes to exist")
```

Comment:
just for a potential must-gather in the future, you may want to also supply what nodes were found already